lucien-ios/AddNewComicBook/AddComicViewController.swift

```diff
@@ -276,39 +314,47 @@ final class AddComicViewController: UIViewController, UIPickerViewDelegate, UIPi
     }
 
     func textFieldShouldReturn(_ textField: UITextField) -> Bool {
-        if let nextTextField = textField.superview?.superview?.viewWithTag(textField.tag + 1) as? UITextField {
+        guard let textFields = addComicViewControllerTextFields else { return false }
+        guard let currentTextFieldArrayIndex = textFields.index(of: textField) else { return false }
+        let currentTextFieldIndex = textFields.startIndex.distance(to: currentTextFieldArrayIndex)
+        if currentTextFieldIndex < textFields.count {
+            let nextTextField = textFields[currentTextFieldIndex + 1]
             nextTextField.becomeFirstResponder()
-        } else {
-            textField.resignFirstResponder()
```

Comment:
Why does this not need to resign anymore?

Reply:
That was an accident. I'll bring it back.